annot_test.go

```diff
@@ -354,15 +406,16 @@ func TestWrite(t *testing.T) {
 ↑
 └─ line1
 `,
-			wantErr: false,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			w := &bytes.Buffer{}
 			err := Write(w, tt.annots...)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("Write() error = %v, wantErr %v", err, tt.wantErr)
+			if tt.wantErr != nil {
+				if tt.wantErr.Error() != err.Error() {
```

Comment:
I would have used

```suggestion
				if !errors.Is(err, wantErr) {
```

Reply:
Thanks for your comment. I tried it first with `errors.Is()` and it didn't work. Now I figured out that I need to implement an `Is()` method for the custom errors.

Reply:
I don't think you have, but also should, do that.

The right way is to add a Unwrap method to your custom error

Look at least the implementation

https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/errors/wrap.go;l=44

So you should add an Unwrap and use errors.Is here

Reply:
What error should be unwrapped if there is no inner error?

`errors.Is()` is used in the latest change. Maybe it is not the right way to implement `Is()` method. But how does it work instead?

Reply:
you return nil then